### ISSUE TITLE
proposal: `dart_style_todos`

### ISSUE THREAD

**Issue by HosseinYousefi** [dart-lang/sdk#59539]
## `dart_style_todos`

Similar to [`flutter_style_todos`](https://dart.dev/tools/linter-rules/flutter_style_todos) but for repos in `dart-lang` org.

People could not longer be working on a project and taking the original context of the TODO with them. Enforcing to have an accompanying link that explains the issue in full detail helps eliminate this problem. 

Links are better than GitHub issue numbers:
* They are clickable from more editors than issue numbers. 
* Sometimes the issue is not on GitHub.
* Links survive migrations to the other repos.

## Description

Use Dart TODO format: // TODO(https://URL-to-issue): message.

## Details

TODOs should include the string TODO in all caps, followed by the link to the respective issue in parenthesis.

## Kind

Style

## Bad Examples

```
// TODO: xyz.
// TODO(name): xyz.
// TODO(#123): xyz.
```

## Good Examples 

```
// TODO(https://github.com/dart-lang/linter/issues/123): xyz.
```

## Discussion

https://github.com/dart-lang/native/issues/1522

@devoncarew @mosuem @pq We could add this lint to https://github.com/dart-lang/lints to be consistent everywhere.

**Comment by mosuem** [comment]
This is the sort of lint which I would have no problem testing for before submitting code, but would be very annoying while developing code, where the `TODO`s are more for me personally. I don't know if there is a way of having both these things.

**Comment by HosseinYousefi** [comment]
> This is the sort of lint which I would have no problem testing for before submitting code, but would be very annoying while developing code, where the `TODO`s are more for me personally. I don't know if there is a way of having both these things.

I use `FIXME`s when developing code.

**Comment by bwilkerson** [comment]
No, we don't currently have any way to specify when a lint should be run (always, before committing, etc.). We also don't have any hooks to allow us to run a lint when you're about to commit some code.

**Comment by srawlins** [comment]
We won't be implementing this as a warning or a lint rule in the Dart analyzer. We now have a high bar for implementing built-in static analysis rules. As of Dart 3.10 (Flutter 3.38), a [new analyzer plugins system](https://dart.dev/tools/analyzer-plugins) is available, and the requested static analysis here could be implemented as an analyzer plugin.
